src/main/java/moriyashiine/aylyth/mixin/LivingEntityMixin.java

```diff
@@ -6,27 +6,120 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.attribute.EntityAttributes;
 import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.entity.effect.StatusEffect;
 import net.minecraft.entity.effect.StatusEffectInstance;
 import net.minecraft.entity.effect.StatusEffects;
 import net.minecraft.entity.mob.MobEntity;
 import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.ItemScatterer;
 import net.minecraft.world.World;
+import org.jetbrains.annotations.Nullable;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.*;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
+import java.util.Arrays;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
 @Mixin(LivingEntity.class)
 public abstract class LivingEntityMixin extends Entity {
 	@Shadow public int deathTime;
 
+	@Shadow public abstract int getArmor();
+
+	@Shadow @Nullable public abstract StatusEffectInstance getStatusEffect(StatusEffect effect);
+
+	@Shadow public abstract boolean hasStatusEffect(StatusEffect effect);
+
+	@Shadow public abstract void setHealth(float health);
+
+	@Shadow public abstract float getHealth();
+
+	@Shadow public abstract boolean isHolding(Item item);
+
+	@Shadow public abstract boolean addStatusEffect(StatusEffectInstance effect);
+
+	@Shadow public abstract boolean isHolding(Predicate<ItemStack> predicate);
+
 	public LivingEntityMixin(EntityType<?> type, World world) {
 		super(type, world);
 	}
+	@ModifyVariable(method = "damage", at = @At("HEAD"), argsOnly = true)
+	float damage(float value, DamageSource source) {
```

Comment:
I would appreciate if anything that does not affect the return value could instead be written in its own `Blighted/Vampiric...` item class using `postHit` to, for example, apply effects and such.

Reply:
Tried adding a helper class and removing a considerable amount of the returns, behavior should be the same